src/torchcodec/samplers/_implem.py

```diff
@@ -25,21 +24,36 @@ def _validate_params(
             f"num_indices_between_frames ({num_indices_between_frames}) must be strictly positive"
         )
 
+    if policy not in _POLICY_FUNCTIONS.keys():
+        raise ValueError(
+            f"Invalid policy ({policy}). Supported values are {_POLICY_FUNCTIONS.keys()}."
+        )
+
 
 def _validate_sampling_range(
-    *, sampling_range_start, sampling_range_end, num_frames, clip_span
+    *,
+    num_indices_between_frames,
+    num_frames_per_clip,
+    sampling_range_start,
+    sampling_range_end,
+    num_frames_in_video,
 ):
     if sampling_range_start < 0:
-        sampling_range_start = num_frames + sampling_range_start
+        sampling_range_start = num_frames_in_video + sampling_range_start
 
-    if sampling_range_start >= num_frames:
+    if sampling_range_start >= num_frames_in_video:
         raise ValueError(
             f"sampling_range_start ({sampling_range_start}) must be smaller than "
-            f"the number of frames ({num_frames})."
+            f"the number of frames ({num_frames_in_video})."
         )
 
+    clip_span = _get_clip_span(
+        num_indices_between_frames=num_indices_between_frames,
+        num_frames_per_clip=num_frames_per_clip,
+    )
+
     if sampling_range_end is None:
-        sampling_range_end = num_frames - clip_span + 1
+        sampling_range_end = max(num_frames_in_video - clip_span + 1, 1)
```

Comment:
Based on the discussion below, shouldn't this just be `num_frames_in_video`? That is, the default last index at which a clip can start is the final frame in the video. And if we're worried that `num_frames_in_video` is not at least 1, maybe we should explicitly check for (and error?) on that?

Even if we decide line 56 is correct, I can't reconcile it and 67. There, it looks like we're making sure that `sampling_range_end` is not larger than the video itself - which makes sense to me. But there, we don't subtract off the size of a clip, but here at line 56, we do.

Reply:
I think the lack of documentation is making this significantly harder to review, sorry about that.

So, my reasoning is the following:

- what happens at line 56 is the default behavior. In line 67, this isn't the default behavior anymore,  `sampling_range_end` is passed by the user
- the default behavior (56) is: "automatically set the upper bound of where a clip can start to a value such that we never try to sample beyond the last frame."
- the reason this is the default behavior is because there is *no* way for users to explicitly set such value, unless they do the match of computing the `clip_span` (and we don't want them to do that).
- Now, at line 67 when the user explicitly passes `sampling_range_end`, we just make sure it doesn't go above `num_frames_in_video` to avoid errors on short videos. If we were to substract `clip_span` there, it would change the semantic of `sampling_range_end` from "upper bound of the start" to "upper bound of the end"

Reply:
Yes, that all tracks. I _think_ that is consistent with what I surmised below, when I read the test.

Reply:
SG, thanks. I don't have a super strong preference either. Let me merge now to unblock, but happy to revisit.